akka-serialization-jackson/src/main/scala/akka/serialization/jackson/JacksonSerializer.scala

```diff
@@ -0,0 +1,369 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.serialization.jackson
+
+import java.io.ByteArrayInputStream
+import java.io.ByteArrayOutputStream
+import java.io.NotSerializableException
+import java.util.zip.GZIPInputStream
+import java.util.zip.GZIPOutputStream
+
+import scala.annotation.tailrec
+import scala.util.Failure
+import scala.util.Success
+import scala.util.control.NonFatal
+
+import akka.actor.ExtendedActorSystem
+import akka.annotation.InternalApi
+import akka.event.LogMarker
+import akka.event.Logging
+import akka.serialization.BaseSerializer
+import akka.serialization.SerializationExtension
+import akka.serialization.SerializerWithStringManifest
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.jsontype.impl.SubTypeValidator
+import com.fasterxml.jackson.dataformat.cbor.CBORFactory
+import com.fasterxml.jackson.dataformat.smile.SmileFactory
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] object JacksonSerializer {
+
+  /**
+   * Using the blacklist from Jackson databind of class names that shouldn't be allowed.
+   * Not nice to depend on implementation details of Jackson, but good to use the same
+   * list to automatically have the list updated when new classes are added in Jackson.
+   */
+  class GadgetClassBlacklist extends SubTypeValidator {
+
+    private def defaultNoDeserClassNames: java.util.Set[String] =
+      SubTypeValidator.DEFAULT_NO_DESER_CLASS_NAMES // it's has protected visibility
+
+    private val prefixSpring: String = "org.springframework."
+    private val prefixC3P0: String = "com.mchange.v2.c3p0."
+
+    def isAllowedClassName(className: String): Boolean = {
+      if (defaultNoDeserClassNames.contains(className))
+        false
+      else if (className.startsWith(prefixC3P0) && className.endsWith("DataSource"))
+        false
+      else
+        true
+    }
+
+    def isAllowedClass(clazz: Class[_]): Boolean = {
+      if (clazz.getName.startsWith(prefixSpring)) {
+        isAllowedSpringClass(clazz)
+      } else
+        true
+    }
+
+    @tailrec private def isAllowedSpringClass(clazz: Class[_]): Boolean = {
+      if (clazz == null == clazz.equals(classOf[java.lang.Object]))
```

Comment:
Is that meant:

```suggestion
      if (clazz == null || clazz.equals(classOf[java.lang.Object]))
```

Reply:
yes, thanks. funny that it's even valid syntax